### PR TITLE
Dropdown article

### PR DESCRIPTION
**Pull request by VelvetThePanda**
# Summary
Adds dropdown/selectmenu article to the docs

# Details
Defines how to create dropdowns, and outlines some limitations.

# Changes proposed
- Fix infinite recursion in interactivity
- Add dropdown article

# Notes
It's the day you've all been waiting for @kuylar @IDoEverything @DWaffles

### CODE REVIEW

File: DSharpPlus/Entities/Interaction/Components/DiscordSelectComponentOption.cs [hunk 1]
```diff
@@ -66,6 +67,25 @@ public DiscordSelectComponentOption(string label, string value, string descripti
...
+                new DiscordSelectComponentOption("Label, Description, Emoji (Default selection)", "label_with_desc_emoji_default", "This is a description!", isDefault: true, new DiscordComponentEmoji(854260064906117121))
+            };
+
+// Make the dropdown
+            var dropdown = new DiscordSelectComponent("dropdown", null, options, false, 1, 2);
+
+// [...] Code trunctated for brevity
+
+            var builder = new DiscordMessageBuilder().WithContent("Look, it's a dropdown!").AddComponents(dropdown);
+
```
Comment (IDoEverything) [inline DSharpPlus/Entities/Interaction/Components/DiscordSelectComponentOption.cs:88]: why was this added to the constructor

Comment (VelvetThePanda) [reply DSharpPlus/Entities/Interaction/Components/DiscordSelectComponentOption.cs:88]: Why was what added

File: docs/articles/advanced_topics/selects.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+"But you skipped options!", you may say, and that we have. Options are a bit more complicated, and have their own section right below.
+
+
+# Dropdown options
+
+Dropdown options are somewhat more involved than handling buttons, but they're still relatively simple.
+
+Dropdowns can have up to **25** options, but must have at least 1.
+
+Dropdown articles consist of several parts:
```
Comment (IDoEverything) [inline docs/articles/advanced_topics/selects.md:54]: Dropdown options*

**Review by IDoEverything — CHANGES_REQUESTED** [review]

File: DSharpPlus/Entities/Interaction/Components/DiscordSelectComponentOption.cs [hunk 3]
```diff
@@ -20,6 +20,7 @@
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
+using System.Collections.Generic;
```
Comment (IDoEverything) [inline DSharpPlus/Entities/Interaction/Components/DiscordSelectComponentOption.cs:23]: remove this too

**Review by IDoEverything — APPROVED** [review]
